rmf_task_ros2/include/rmf_task_ros2/bidding/Auctioneer.hpp

```diff
@@ -80,6 +89,47 @@ class Auctioneer : public std::enable_shared_from_this<Auctioneer>
     BiddingResultCallback result_callback,
     ConstEvaluatorPtr evaluator);
 
+  /// Create an instance of the Auctioneer. This instance will handle all
+  /// the task dispatching bidding mechanism. A default evaluator is used.
+  ///
+  /// \param[in] node_base_interface
+  ///   ros2 node base interface for managing callback groups and timers
+  ///
+  /// \param[in] node_clock_interface
+  ///   ros2 node clock interface for managing time
+  ///
+  /// \param[in] node_logging_interface
+  ///   ros2 node logging interface for enabling logging
+  ///
+  /// \param[in] node_timers_interface
+  ///   ros2 node timers interface for adding timers
+  ///
+  /// \param[in] node_topics_interface
+  ///   ros2 node topics interface for creating publishers and subscribers
+  ///
+  /// \param[in] node_parameters_interface
+  ///   ros2 node parameters interface for handling parameters
+  ///
+  /// \param[in] result_callback
+  ///   This callback fn will be called when a bidding result is concluded
+  ///
+  /// \sa make()
+  static std::shared_ptr<Auctioneer> make(
+    const rclcpp::node_interfaces::NodeBaseInterface::SharedPtr
+    node_base_interface,
+    const rclcpp::node_interfaces::NodeClockInterface::SharedPtr
+    node_clock_interface,
+    const rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr
+    node_logging_interface,
+    const rclcpp::node_interfaces::NodeTimersInterface::SharedPtr
+    node_timers_interface,
+    const rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr
+    node_topics_interface,
+    const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr
+    node_parameters_interface,
+    BiddingResultCallback result_callback,
+    ConstEvaluatorPtr evaluator);
```

Comment:
Suggestion for a cleaner API. If we don't need all the node interfaces we can replace `ALL_RCLCPP_NODE_INTERFACES` with a list of interfaces we need.

```suggestion
  using ActioneerInterfaces = rclcpp::node_interfaces::NodeInterfaces<ALL_RCLCPP_NODE_INTERFACES>
  
  static std::shared_ptr<Auctioneer> make(
     ActioneerInterfaces node_interfaces,
    BiddingResultCallback result_callback,
    ConstEvaluatorPtr evaluator);
```

Reply:
Sweet! Yeah we don't need all of them, I've already listed out the ones that we need